src/main/java/com/kodilla/ecommercee/domain/ProductItem.java

```diff
@@ -0,0 +1,27 @@
+package com.kodilla.ecommercee.domain;
+
+import com.kodilla.ecommercee.domain.dto.ProductDto;
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+
+import javax.persistence.GeneratedValue;
+import javax.persistence.Id;
+import javax.persistence.OneToMany;
+import java.math.BigDecimal;
+import java.util.List;
+
+@AllArgsConstructor
+@NoArgsConstructor
+@Getter
+public class ProductItem {
```

Comment:
Nie do końca rozumiem zastosowanie tej klasy i dlaczego encja Cart przechowuje liste tych obiektów ? 

W specyfikacji projektu  jest zaznaczone aby Cart przechowywał listę  obiektów typu Product : https://github.com/interceptor2048/project-jdp-1907-01/blob/master/specs/cart.spec.json
Więc jakie jest szczegółowe zastosowanie klasy ProductItem bo nie rozumiem chyba zabardzo ?

Reply:
To co jest w koszyku w obiekcie ProductItem zawiera Listę produktów (List) które pochodzą z bazy danych,oraz dodatkowe informacje, których nie ma w klasie Product, czyli BigDecimal ammount i int quantity. Nie ma sensu rozszerzać klasy Product o te dodatkowe pola w prawdziwym sklepie koszyk zawiera listę produktów oraz dodatkowe informacje.